### PR TITLE
Remove symlink to crd file in kubernetes e2e test data

### PR DESCRIPTION
**Pull request by sheidkamp**
# Description
A symlink in the test data causes test failure/file-not-found when imported into `solo-projects`

## Code changes
Added `Directory()` to the `projects/gateway2/crds` package to provide a way to access the crd files.

Removed symlink and updated `types.go` to point to the original file using `crds.Directory()`

## Testing steps
Locally, test still passes:
```
ISTIO_VERSION=1.21.0  GOLANG_PROTOBUF_REGISTRATION_CONFLICT=ignore go test -v -timeout 600s ./test/kubernetes/e2e/tests -run ^TestK8sGateway$/^HTTPRouteServices$/^TestConfigureHTTPRouteBackingDestinationsWithServiceAndWithoutTCPRoute$
```

```
...
  Gloo was successfully uninstalled.
--- PASS: TestK8sGateway (113.27s)
    --- PASS: TestK8sGateway/HTTPRouteServices (46.08s)
        --- PASS: TestK8sGateway/HTTPRouteServices/TestConfigureHTTPRouteBackingDestinationsWithServiceAndWithoutTCPRoute (46.08s)
```

Validated by pulling into solo-projects:
```
go get github.com/solo-io/gloo@remove-manifest-symlink
```
and running tests on an existing cluster:
```
go clean -testcache; GOLANG_PROTOBUF_REGISTRATION_CONFLICT=ignore go test -v -timeout 600s ./test/kubernetes/e2e/tests -run  ^TestOssFeatures$/^OssK8sGateway$/^HTTPRouteServices$/^TestConfigureHTTPRouteBackingDestinationsWithServiceAndWithoutTCPRoute$
```
With output:
```
....
  Gloo was successfully uninstalled.
--- PASS: TestOssFeatures (100.21s)
    --- PASS: TestOssFeatures/OssK8sGateway (100.21s)
        --- PASS: TestOssFeatures/OssK8sGateway/HTTPRouteServices (48.20s)
            --- PASS: TestOssFeatures/OssK8sGateway/HTTPRouteServices/TestConfigureHTTPRouteBackingDestinationsWithServiceAndWithoutTCPRoute (48.20s)
PASS
```

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by andy-fong — APPROVED** [review]

File: test/kubernetes/e2e/features/services/httproute/types.go [hunk 1]
```diff
@@ -15,9 +15,12 @@ import (
 )
 
 var (
+	// This is fragile, but symlinks don't work when the tests are imported to another repo, and its better than duplicating the file
+	repoRoot = filepath.Join(util.MustGetThisDir(), "..", "..", "..", "..", "..", "..")
```
Comment (sam-heilbron) [inline test/kubernetes/e2e/features/services/httproute/types.go:19]: https://github.com/solo-io/gloo/blob/d613a7d53616ab6b34144291bf4df69708728bb4/test/kubernetes/e2e/features/helm_settings/suite.go#L57 

Can we use GetModuleRoot instead? 

Alternatively, we could define a method in the crds/ package that returns the full path to a file, and then call it here. that way the functionalty lives in the package itself and others can use

Comment (sheidkamp) [reply test/kubernetes/e2e/features/services/httproute/types.go:19]: That does not work when imported into `solo-projects`

Comment (sheidkamp) [reply test/kubernetes/e2e/features/services/httproute/types.go:19]: hmm, and what I tried was `GetGoModule` not `GetModuleRoot`, so let me try that too

Comment (sam-heilbron) [reply test/kubernetes/e2e/features/services/httproute/types.go:19]: If we exposed a method in the crds package that usesd "GetThisDir/file-name", couldn't that work?

**Review by sam-heilbron — APPROVED** [review]

**Pull request merged**